### DISCUSSION TITLE
Mysterious PV Curtailment in Outage

### DISCUSSION THREAD

**Discussion by adam-morse** [NREL/REopt-API-Analysis#153] [4 - Resilience Questions]
Hello ReOpt,

I am validating ReOpt's results against another techno-economic software and am seeing behavior in resilient operation that I can't explain. My intent in the exercise is--with PV size fixed--have ReOpt compute the required battery size to get through an outage. I want to find the minimum battery size feasible to serve the critical load (no other generation included). 

The results show me ReOpt sizing battery energy over 2x of the other software and I'm unclear why this is happening. What has me stumped is why ReOpt is curtailing the PV in the outage window. 

I wonder if not being able to capture the full PV is somehow preventing it from a more optimal sizing. Given tha the battery AC sizing was coming in very low (~100kw). I forced the inverter sizing sufficiently high to swallow the max PV but still saw the same curtailment.

I'm attaching the inputs and outputs from the exercise. The parameters I'm using are 1.5GWh annual load (30% is critical for the outage), 500kW PV, trying to get through one month summer outage.

I am expecting battery size in the 700kWh ballpark but it's spitting out 1.7MWh.

Thanks,
Adam

[evaluation_inputs.json](https://github.com/NREL/REopt-Analysis-Scripts/files/15155082/evaluation_inputs.json)
[evaluation_results.json](https://github.com/NREL/REopt-Analysis-Scripts/files/15155083/evaluation_results.json)

**Comment by kurt-verdant** [comment]
To me it looks like during the 38 hours 4962-4999,

sum PV gen              =   814 kWh
sum critical loads = 2,136 kWh
difference                = 1,322 kWh

If the battery RTE is 0.96, that would put battery size at 1,377 kWh (>>700 kWh)

Then a question is: Why is the REopt result 1,757 kWh?

My hunch is the larger battery minimizes NPV

The MILP's objective function minimizes energy lifecycle cost (i.e., capital costs, O&M costs, utility costs, and emissions costs). When critical loads are relatively small (30%), it wouldn't surprise me if utility costs are driving the result. I didn't check the tariff, but if there are demand charges, they might explain why the final answer is larger than what resilience alone might suggest. 

Or maybe my math above isn't complete. Maybe 1,322 kWh needs to be increased to account for soc_min_fraction = 0.2. Actually this is probably a more likely explanation than the NPV hunch.

Maybe REopt is using different weather or different tariff than the other tools it's being compared to.

I peeked at this problem because I just started using REopt and I thought it would be a good way to learn. In case any other new people see this in the future I'll share one thing that tripped me up for a minute. In the "Outages" key in the results there are numerous kW series and they all need to be considered because for outage hours they essentially replace other values available in 8760 series in PV, ElectricStorage, etc. The other thing I found helpful when I ran into questions about how to write input JSON was to load up a Web Tool run with similar specs, then pull pieces of JSON out of the input file available for the Web Tool run.
